client/src/queries/artists.ts

```diff
@@ -0,0 +1,18 @@
+import { QueryFunction, queryOptions } from "@tanstack/react-query";
+import * as api from "./fetch/fetchWrapper";
+
+const fetchArtist: QueryFunction<
```

Comment:
Yeah - it is a bit verbose, but I figured that might have its advantages as well if any specific behavior needs to be added in the future.

On other projects I've used code generators such as [openapi-generator.tech](https://openapi-generator.tech) to automate the types/functions for larger REST APIs. I've found that this can work really well IF there is complete coverage on both sides of the API - i.e. the server implementation generates an accurate OpenAPI spec which then generates a fully-compatible interface for the client.

It can be a bit kludgey to set up, though, and IMO it loses its benefits if any part of that needs to be written manually - so I'm not sure if that would be worth the effort here.

There *are* a few less config-heavy alternatives to the code generation approach that guarantee type-safety: [NextJS Server Actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) and [tRPC](https://trpc.io) are two that I can think of. I have yet to try either, though - and I wouldn't want to migrate an entire existing project to them, especially when it's already working fine! So I'm resisting the "ooh shiny!" urge lol